internal/drivers/azure/driver.go

```diff
@@ -0,0 +1,344 @@
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes(c.size)),
+			},
+			StorageProfile: &armcompute.StorageProfile{
+				ImageReference: &armcompute.ImageReference{
+					Publisher: to.Ptr(c.publisher),
+					Offer:     to.Ptr(c.offer),
+					SKU:       to.Ptr(c.sku),
+					Version:   to.Ptr(c.version),
+				},
+				OSDisk: &armcompute.OSDisk{
+					Name:         to.Ptr(diskName),
+					CreateOption: to.Ptr(armcompute.DiskCreateOptionTypesFromImage),
+					Caching:      to.Ptr(armcompute.CachingTypesReadWrite),
+					ManagedDisk: &armcompute.ManagedDiskParameters{
+						StorageAccountType: to.Ptr(armcompute.StorageAccountTypesStandardLRS), // OSDisk type Standard/Premium HDD/SSD
+					},
+				},
+			},
+			OSProfile: &armcompute.OSProfile{ //
+				ComputerName:  to.Ptr("vm-runner"),
+				AdminUsername: to.Ptr(c.username),
+				AdminPassword: to.Ptr(c.password),
+				CustomData:    to.Ptr(uData),
+			},
+			NetworkProfile: &armcompute.NetworkProfile{
+				NetworkInterfaces: []*armcompute.NetworkInterfaceReference{
+					{
+						ID: to.Ptr(*networkInterface.ID),
+					},
+				},
+			},
+		},
+	}
+
+	if opts.OS == oshelp.OSWindows {
+		in.Plan = &armcompute.Plan{
+			Name:      to.Ptr(c.offer),
+			Product:   to.Ptr(c.offer),
+			Publisher: to.Ptr(c.publisher),
+		}
+	}
+
+	poller, err := c.service.BeginCreateOrUpdate(ctx, c.resourceGroupName, name, in, nil)
+	if err != nil {
+		return nil, err
+	}
+	vm, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	logr.Debugln("instance insert operation completed")
+
+	logr.
+		WithField("ip", vm.ID).
+		WithField("time", fmt.Sprintf("%.2fs", time.Since(startTime).Seconds())).
+		Debugln("azure: [provision] VM provisioned")
+
+	instanceMap := c.mapToInstance(&vm, opts)
+	logr.
+		WithField("ip", instanceMap.Address).
+		WithField("time", fmt.Sprintf("%.2fs", time.Since(startTime).Seconds())).
+		Debugln("azure: [provision] complete")
+
+	return &instanceMap, nil
+}
+
+func (c *config) Destroy(ctx context.Context, instanceIDs ...string) (err error) {
+	logr := logger.FromContext(ctx)
+	if c.resourceGroupName == "" {
+		c.resourceGroupName = defaultResourceGroup
+	}
+	logr.Debugln("Azure destroy operation started")
+	if len(instanceIDs) == 0 {
+		return
+	}
+	for _, instanceID := range instanceIDs {
+		vnetName := fmt.Sprintf("%s-vnet", instanceID)
+		publicIPName := fmt.Sprintf("%s-publicip", instanceID)
+		networkInterfaceName := fmt.Sprintf("%s-networkinterface", instanceID)
+		diskName := fmt.Sprintf("%s-disk", instanceID)
+
+		logr.Debugln("Azure destroying instance", instanceID)
+		logr.WithField("id", instanceID).
+			WithField("cloud", types.Google)
+
+		poller, err := c.service.BeginDelete(ctx, c.resourceGroupName, instanceID, nil)
+		if err != nil {
+			return err
+		}
+		_, err = poller.PollUntilDone(ctx, nil)
+		if err != nil {
+			return err
+		}
+		logr.Debugf("Azure instance destroyed %s", instanceID)
```

Comment:
just leaving it here: maybe better to convert debug logs to info logs if you feel they are insightful

Reply:
yeah changed.